presto-main/src/main/java/io/prestosql/cost/PlanNodeStatsEstimate.java

```diff
@@ -92,26 +91,22 @@ private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics, Type
     {
         checkArgument(type != null, "type is null");
 
-        double averageRowSize = symbolStatistics.getAverageRowSize();
         double nullsFraction = firstNonNaN(symbolStatistics.getNullsFraction(), 0d);
         double numberOfNonNullRows = outputRowCount * (1.0 - nullsFraction);
 
-        if (isNaN(averageRowSize)) {
-            if (type instanceof FixedWidthType) {
-                averageRowSize = ((FixedWidthType) type).getFixedSize();
-            }
-            else {
-                averageRowSize = DEFAULT_DATA_SIZE_PER_COLUMN;
-            }
-        }
-
-        double outputSize = numberOfNonNullRows * averageRowSize;
+        double outputSize = 0;
 
         // account for "is null" boolean array
-        outputSize += outputRowCount * Byte.BYTES;
+        outputSize += outputRowCount;
+
+        if (type instanceof FixedWidthType) {
+            outputSize += numberOfNonNullRows * ((FixedWidthType) type).getFixedSize();
+        }
+        else {
+            double averageRowSize = firstNonNaN(symbolStatistics.getAverageRowSize(), DEFAULT_DATA_SIZE_PER_COLUMN);
+            outputSize += numberOfNonNullRows * averageRowSize;
 
-        // account for offsets array for variable width types
-        if (type instanceof VariableWidthType) {
+            // account for offsets array
```

Comment:
this isn't accurate for other types, e.g: `RowType`. Add `TODO`

Reply:
added